src/script/interpreter.cpp

```diff
@@ -1416,9 +1416,26 @@ bool GenericTransactionSignatureChecker<T>::CheckSequence(const CScriptNum& nSeq
 template class GenericTransactionSignatureChecker<CTransaction>;
 template class GenericTransactionSignatureChecker<CMutableTransaction>;
 
+static bool ExecuteWitnessProgram(std::vector<std::vector<unsigned char>>&& stack, const CScript& scriptPubKey, unsigned int flags, SigVersion sigversion, const BaseSignatureChecker& checker, ScriptError* serror)
```

Comment:
Hi 🙋 I'd also like to bikeshed this function name please.

My understanding is that the witness program is the 20/32 byte commitment to the spending condition. The 'witness script' is what is actually executed. Does `ExecuteWitnessScript()` makes more sense as a function name? For both P2WPKH and P2WSH, the program is not actually passed in here.

Reply:
Bikeshed accepted.